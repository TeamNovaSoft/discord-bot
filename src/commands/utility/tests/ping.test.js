const pingCommand = require('../ping');

jest.mock('../../../languages', () => ({
  translateLanguage: jest.fn((key) => {
    if (key === 'ping.description') {
      return 'Ping command description';
    }
    return key;
  }),
  keyTranslations: jest.fn((key) => {
    if (key === 'ping.description') {
      return {
        'en-US': 'Ping command description',
        'es-ES': 'DescripciÃ³n del comando Ping',
      };
    }
    return {};
  }),
}));

describe('Ping command', () => {
  test('should have the correct name and description', () => {
    const commandJSON = pingCommand.data.toJSON();
    expect(commandJSON.name).toBe('ping');
    expect(commandJSON.description).toBe('Ping command description');
  });

  test('command snapshot', () => {
    const commandJSON = pingCommand.data.toJSON();
    expect(commandJSON).toMatchSnapshot();
  });

  test('should respond with "Pong!" when executed', async () => {
    const interaction = {
      reply: jest.fn().mockResolvedValue(true),
    };

    await pingCommand.execute(interaction);
    expect(interaction.reply).toHaveBeenCalledWith('Pong!');
  });
});
