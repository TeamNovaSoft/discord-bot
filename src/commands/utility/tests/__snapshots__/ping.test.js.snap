// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ping command command snapshot 1`] = `
{
  "contexts": undefined,
  "default_member_permissions": undefined,
  "default_permission": undefined,
  "description": "Ping command description",
  "description_localizations": undefined,
  "dm_permission": undefined,
  "integration_types": undefined,
  "name": "ping",
  "name_localizations": undefined,
  "nsfw": undefined,
  "options": [],
  "type": 1,
}
`;
